LibreNMS/OS/Endrun.php

```diff
@@ -34,15 +34,14 @@ class Endrun extends OS implements OSDiscovery
     public function discoverOS(Device $device): void
     {
         if (Str::contains($device->sysDescr, 'Sonoma')) {
-            $info = snmp_get_multi($this->getDeviceArray(), ['gntpVersion.0', 'snmpSetSerialNo.0'], '-OQUs', 'SONOMA-MIB:SNMPv2-MIB');
+            $info = snmp_get_multi($this->getDeviceArray(), ['gntpVersion.0', 'snmpSetSerialNo.0'], '-OQUs', 'SONOMA-MIB:MERIDIAN2-MIB:SNMPv2-MIB');
```

Comment:
If both contain gntpVersion or snmpSetSerialNo, it will only resolve the oid from the first listed.  It will not try one than the other.

Reply:
Might be easier to implement this in yaml and call `parent::discoverOS($device)`  You can list both separately and it will try one then the other.

Reply:
All this regex stuff and yaml. this ought to be fun. Thanks, will try and see how far I get.